operatorcontrolplane/v1alpha1/types_conditioncheck.go

```diff
@@ -0,0 +1,150 @@
+// Package v1alpha1 is an API version in the controlplane.operator.openshift.io group
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// PodNetworkConnectivityCheck
+type PodNetworkConnectivityCheck struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	// Spec defines the source and target of the connetivity check
+	// +kubebuilder:validation:Required
+	// +required
+	Spec PodNetworkConnectivityCheckSpec `json:"spec"`
+
+	// Status contains the observed status of the connectivity check
+	// +optional
+	Status PodNetworkConnectivityCheckStatus `json:"status"`
+}
+
+type PodNetworkConnectivityCheckSpec struct {
+	// SourcePod names the pod from which the condition will be checked
+	// +kubebuilder:validation:Required
+	// +required
+	SourcePod string `json:"sourcePod"`
+
+	// EndpointAddress to check. A TCP address of the form host:port. Note that
+	// if host is a DNS name, then the check would fail if the DNS name cannot
+	// be resolved. Specify an IP address for host to bypass DNS name lookup.
+	// +kubebuilder:validation:Required
+	// +required
+	TargetEndpoint string `json:"targetEndpoint"`
+}
+
+// +k8s:deepcopy-gen=true
+type PodNetworkConnectivityCheckStatus struct {
+	// Successes contains logs successful check actions
+	// +optional
+	Successes []LogEntry `json:"successes"`
+
+	// Failures contains logs of unsuccessful check actions
+	// +optional
+	Failures []LogEntry `json:"failures"`
+
+	// Outages contains logs of time periods of outages
+	// +optional
+	Outages []OutageEntry `json:"outages"`
+
+	// Conditions summarize the status of the check
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	// +optional
+	Conditions []PodNetworkConnectivityCheckCondition `json:"conditions" patchStrategy:"merge" patchMergeKey:"type"`
+}
+
+// LogEntry records events
+type LogEntry struct {
+	// Start time of check action.
+	Start metav1.Time `json:"time,omitempty"`
+
+	// Status indicates if the action performed by the check was successful or not.
+	// +required
+	Success bool `json:"status"`
+
+	// Reason for status in a machine readable format.
+	// +optional
+	Reason string `json:"reason,omitempty"`
+
+	// Message explaining status in a human readable format.
+	// +optional
+	Message string `json:"message,omitempty"`
+
+	// Latency records how long the action mentioned in the entry took.
+	// +optional
+	Latency metav1.Duration `json:"latency"`
```

Comment:
this must be nullable because `Duration` has a custom marshaller.

Reply:
Updated.